### PR TITLE
copr: add non-vectorize JsonValidJson

### LINKED ISSUE #6798: UCP: Migrate scalar function `JsonValidJsonSig` from TiDB
## Description

Port the scalar function `JsonValidJsonSig` from TiDB to coprocessor.

## Score 

* 50

## Mentor(s)

* @lonng 

## Recommended Skills

* Rust programming 

## Learning Materials

Already implemented expressions ported from TiDB

- https://github.com/tikv/tikv/tree/master/components/tidb_query/src/rpn_expr) 
- https://github.com/tikv/tikv/tree/master/components/tidb_query/src/expr)

### PR DESCRIPTION
**Pull request by return02**
Signed-off-by: aooohooo <aooohooo123@gmail.com>

UCP: #6798 

Checklist: #5751 

### What problem does this PR solve?

Issue Number: close #6798  

Problem Summary: add non-vectorize JsonValidJson

### What is changed and how it works?

What's Changed: Port the scalar function JsonValidJson from TiDB to coprocessor.

### Related changes

- No

### Check List 

Tests 

- Unit test

### Release note

### CODE REVIEW

**Comment by TennyZhuang** [comment]
LGTM

File: components/tidb_query_normal_expr/src/builtin_json.rs [hunk 1]
```diff
@@ -780,4 +792,34 @@ mod tests {
...
+    fn test_json_valid_json() {
+        let cases = vec![
+            (Datum::Json(r#""3""#.parse().unwrap()), Some(1)),
+            (
+                Datum::Bytes(r#""hello world""#.to_string().into_bytes()),
+                Some(0),
+            ),
+            (Datum::F64(1.0), Some(0)),
+            (Datum::I64(-123), Some(0)),
+            (Datum::U64(123), Some(0)),
```
Comment (Fullstop000) [inline components/tidb_query_normal_expr/src/builtin_json.rs:806]: Rest LGTM. Prefer to add the rest `Datum` cases expect `Min` and `Max`

**Comment by andylokandy** [comment]
@return02 Sorry for the delay. Unfortunately non-vec expressions no longer exist, so I'm going to close this PR. 

Sorry about that, and thank you for your contribution.

**Pull request closed without merging**